client/src/pages/InitRoomPage/index.tsx

```diff
@@ -1,33 +1,110 @@
-import { useEffect, useRef, useState } from 'react';
-import { InitRoomPageLayout } from '@pages/InitRoomPage/styles';
+import React, { useEffect, useRef, useState } from 'react';
+import {
+  FooterBox,
+  InitRoomPageLayout,
+  MapBox,
+  MarkerBox,
+  StartButton,
+} from '@pages/InitRoomPage/styles';
 import useCurrentLocation from '@hooks/useCurrentLocation';
+import { ReactComponent as MarkerImage } from '@assets/images/marker.svg';
+import { NAVER_ADDRESS } from '@constants/map';
 
 function InitRoomPage() {
   const mapRef = useRef<HTMLDivElement>(null);
-  const location = useCurrentLocation();
+  const userLocation = useCurrentLocation();
+  const [address, setAddress] = useState<string>(NAVER_ADDRESS);
 
-  // 현재 위치에 맞춰 지도 생성
   useEffect(() => {
-    const initMap = () => {
-      if (!mapRef.current) {
-        return;
-      }
+    if (!mapRef.current) {
+      return;
+    }
 
-      const map = new naver.maps.Map(mapRef.current, {
-        center: new naver.maps.LatLng(location.lat, location.lng),
+    // 좌표 -> 주소 변환 & setAddress
+    const reverseGeocode = (lat: number, lng: number) => {
+      naver.maps.Service.reverseGeocode(
+        {
+          coords: new naver.maps.LatLng(lat, lng),
+          orders: [naver.maps.Service.OrderType.ROAD_ADDR, naver.maps.Service.OrderType.ADDR].join(
+            ','
+          ),
+        },
+        // eslint-disable-next-line consistent-return
+        (status, response) => {
+          if (status !== naver.maps.Service.Status.OK) {
+            alert('주소 변환 실패');
+          }
+          setAddress(response.v2.address.roadAddress || response.v2.address.jibunAddress);
+        }
+      );
+    };
```

Comment:
해당 함수 내부에서 setAddress까지 해주는건 너무 많은 일을 하는 것 같다는 생각이 듭니다. 차라리 해당 함수에서는 주소 정보(string)를 반환하는 것으로 끝내고 setAddress 로직은 따로 빼주는 것이 어떨까요?

Reply:
처음에 그렇게 짰었는데요! 
`reverseGeocode` 함수 내에서 변환한 주소 string의 return을 
`naver.maps.Service.reverseGeocode`의 콜백함수 안에서 해주다보니 (37번 line)
타입스크립트가 `reverseGeocode`의 반환값을 void로 추론하는 문제가 생겼습니다.

그러다보니 이 함수 바깥에서 `setAddress(reverseGeocode(위경도값))` 해주려고 했더니 void라서 안된다는 에러가 나고..
이걸 어떻게 해결해야될지 고민하다가 답이 안 떠올라서 그냥 저 안에서 setAddress 처리까지 해버렸어요,,

이 문제를 어떻게 해결하면 좋을까요,,
뭔가 제가 타입스크립트 활용을 제대로 못하고 있는 듯 합니다..ㅎㅎ

Reply:
```suggestion
    const reverseGeocode = async (lat: number, lng: number) => {
    let result:string = '';
      await naver.maps.Service.reverseGeocode(
        {
          coords: new naver.maps.LatLng(lat, lng),
          orders: [naver.maps.Service.OrderType.ROAD_ADDR, naver.maps.Service.OrderType.ADDR].join(
            ','
          ),
        },
        // eslint-disable-next-line consistent-return
        (status, response) => {
          if (status !== naver.maps.Service.Status.OK) {
            alert('주소 변환 실패');
            return;
          }
          result = response.v2.address.roadAddress || response.v2.address.jibunAddress;
        }
      );
      return result;
    };
```
당장 생각나는건 이렇게...? 변수명은 그냥 마음대로 지었습니다.(result)

Reply:
빈 문자열을 리턴했을 땐 무언가 문제가 있는 것으로....

Reply:
로직을 수정하기 어렵다면 함수명이라도 바꾸면 좀 더 나을 것 같아요. updateAddress? 정도로?

Reply:
<img width="880" alt="스크린샷 2022-11-28 오후 7 56 53" src="https://user-images.githubusercontent.com/55318618/204260869-157db3c1-1999-4578-9a43-4bbe9ec69b03.png">

이게 사실 2 -> 1 순서로 출력되어야 원하는대로 동작할텐데 
1 -> 2 순서로 출력이 되어요,, 

<img width="579" alt="스크린샷 2022-11-28 오후 7 57 47" src="https://user-images.githubusercontent.com/55318618/204261080-d4eca6d9-dd9e-4fe7-bbd1-fec0df83cd1e.png">

이걸 어떻게 해야되는지 어렵네요..ㅠㅠ

Reply:
제가 이 코드 로컬에서 돌려보려 했는데 naver maps reverseGeocode 쪽에서 타입에러가 뜨네요?? 
<img width="581" alt="image" src="https://user-images.githubusercontent.com/89074053/204264258-1571bf7b-cf8c-4605-b647-85f9d112acbf.png">

Reply:
읭?? 제가 올린것이요? 아니면 창명님이 짜주신거요!? 
전 둘다 일단 에러가 안뜨긴한데 뭐지

Reply:
환경변수명 수정하신거 제가 추가안해서 오류 나는거 였습니다! 죄송합니다

Reply:
ㅋㅋㅋㅋ괜찮습니당

Reply:
분명 api 요청을 보내는거라 Promise를 반환해야된다고 생각했는데 내부 구조를 까보니 Promise가 아니네요? 그런데 또 작동은 비동기로 하는거 같아서 짜놓으신거에서 구조 수정이 어려워보이긴 합니다.

Reply:
그러면 함수명만 뭔가 setAddress를 포함하는 늬앙스로 고쳐도 될 것 같습니다.

Reply:
끄악.. 그럼 일단 이렇게 하고, 추후 리팩토링 주간에 개선이 가능하다면 고쳐보면 좋을것같아요!!!
적극적인 리뷰.. 감사합니다,,ㅎㅎㅎㅎ